src/posix/platform/netif.cpp

```diff
@@ -505,6 +513,202 @@ static void UpdateLink(otInstance *aInstance)
+    SuccessOrExit(error = AddRtAttr32(&req.header, sizeof(req), RTA_PRIORITY, kExternalRoutePriority));
+    SuccessOrExit(error = AddRtAttr32(&req.header, sizeof(req), RTA_OIF, ifIdx));
+
+    VerifyOrExit(send(sNetlinkFd, &req, sizeof(req), 0) > 0, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+static otError DeleteExternalRoute(const otIp6Prefix &aPrefix)
+{
+    struct
+    {
+        struct nlmsghdr header;
+        struct rtmsg    msg;
+        char            buf[512];
+    } req{};
+    unsigned char data[sizeof(in6_addr)];
+    char          addrBuf[OT_IP6_ADDRESS_STRING_SIZE];
+    unsigned int  ifIdx = otSysGetThreadNetifIndex();
+    otError       error = OT_ERROR_NONE;
+
+    VerifyOrExit(ifIdx > 0, error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(sNetlinkFd >= 0, error = OT_ERROR_INVALID_STATE);
+
+    req.header.nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK | NLM_F_NONREC;
+
+    req.header.nlmsg_len  = NLMSG_LENGTH(sizeof(rtmsg));
+    req.header.nlmsg_type = RTM_DELROUTE;
+    req.header.nlmsg_pid  = 0;
+    req.header.nlmsg_seq  = ++sNetlinkSequence;
+
+    req.msg.rtm_family   = AF_INET6;
+    req.msg.rtm_src_len  = 0;
+    req.msg.rtm_dst_len  = aPrefix.mLength;
+    req.msg.rtm_tos      = 0;
+    req.msg.rtm_scope    = RT_SCOPE_UNIVERSE;
+    req.msg.rtm_type     = RTN_UNICAST;
+    req.msg.rtm_table    = RT_TABLE_MAIN;
+    req.msg.rtm_protocol = RTPROT_BOOT;
+    req.msg.rtm_flags    = 0;
+
+    otIp6AddressToString(&aPrefix.mPrefix, addrBuf, OT_IP6_ADDRESS_STRING_SIZE);
+    inet_pton(AF_INET6, addrBuf, data);
+    SuccessOrExit(error = AddRtAttr(&req.header, sizeof(req), RTA_DST, data, sizeof(data)));
+    SuccessOrExit(error = AddRtAttr32(&req.header, sizeof(req), RTA_OIF, ifIdx));
+
+    VerifyOrExit(send(sNetlinkFd, &req, sizeof(req), 0) > 0, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+bool HasExternalRouteInNetData(otInstance *aInstance, const otIp6Prefix &aExternalRoute)
+{
+    otNetworkDataIterator iterator = OT_NETWORK_DATA_ITERATOR_INIT;
+    otExternalRouteConfig config;
+    bool                  found = false;
+
+    while (otNetDataGetNextRoute(aInstance, &iterator, &config) == OT_ERROR_NONE)
+    {
+        if (otIp6IsAddressEqual(&config.mPrefix.mPrefix, &aExternalRoute.mPrefix) &&
+            config.mPrefix.mLength == aExternalRoute.mLength)
+        {
+            found = true;
+            break;
+        }
+    }
+    return found;
+}
+
+bool HasAddedExternalRoute(const otIp6Prefix &aExternalRoute)
+{
+    bool found = false;
+
+    for (int i = 0; i < static_cast<int>(sAddedExternalRoutesNum); ++i)
+    {
+        if (otIp6IsAddressEqual(&sAddedExternalRoutes[i].mPrefix, &aExternalRoute.mPrefix) &&
+            sAddedExternalRoutes[i].mLength == aExternalRoute.mLength)
+        {
+            found = true;
+            break;
+        }
+    }
+    return found;
+}
+
+static void UpdateExternalRoutes(otInstance *aInstance)
```

Comment:
I wrote this functionality in this file since this file has other functionalities based on netlink. Also, I'd better share the sNetlinkFd with the rest of code. If we want to create a class solely for router management, we may also need to extract shared netlink components and put them in a new class/file.

Reply:
I think it is okay to add the new functions in this file. It sounds good to me to refactor this file to define a class, but it will be a huge change. I think it is not necessary to do it in this PR.

Reply:
Fine to me to leave it as is.